DEVELOP.md

```diff
@@ -0,0 +1,485 @@
+repository.
+
+### Code Style
+
+Daffodil mandates standard Scala formatting and its code generally
+keeps close to that formatting consistently.  No one has run an
+automatic Scala formatter on the codebase yet (there has not been much
+need since the code is already formatted pretty well) but
+[DAFFODIL-2133](https://issues.apache.org/jira/browse/DAFFODIL-2133)
+is waiting for someone willing to do that work.
+
+Daffodil mandates that at least 80% of new code and modified code
+should be covered by unit tests or TDML tests.  Daffodil naming
+conventions and more are covered in the Confluence pages [Code Style
+Guidelines](https://cwiki.apache.org/confluence/display/DAFFODIL/Code+Style+Guidelines)
+and [Coding for
+Performance](https://cwiki.apache.org/confluence/display/DAFFODIL/Coding+for+Performance).
+
+### Code Workflow
+
+It is very important that you fork the Apache repository to your own
+GitHub account before you start making changes to Apache Daffodil
+source code.  You should always make changes only to your own fork of
+the upstream Apache repository, that is, clone your working copy from
+your fork and push commits back to your own fork only.  The same
+workflow applies to Apache Daffodil's other two source code
+repositories, that is, always fork these repositories and clone from
+your forks before you make any changes to the [Daffodil VS Code
+Extension](https://github.com/apache/daffodil-vscode) or [Daffodil
+Website](https://github.com/apache/daffodil-site).  When you are ready
+to contribute code from any of your forks to any of these upstream
+Apache repositories, you must create a pull request from your fork,
+allow the Apache Daffodil developers to review your contribution, make
+any suggested changes, and wait for their approval before you or
+someone else can merge your code into one of Apache's source code
+repositories.
+
+The Daffodil developers want each pull request to add only one commit
+to the Apache repository to keep its git history as readable and
+bisectable as possible.  Your pull request should fix a JIRA issue in
+its entirety but it should not fix 2 or more unrelated JIRA issues in
+the same commit either.  Your pull request should start with only one
+commit in it before you ask the developers to review it and your pull
+request should end with only one commit in it after you receive enough
+reviewer approvals (two +1's for Daffodil and Daffodil VS Code
+Extension, one +1 for Daffodil Website).  If you have made multiple
+commits to your pull request due to reviewer suggestions (which is the
+usual case), you must not merge these multiple commits into the Apache
+repository.  You must rebase and squash your pull request back to one
+commit as shown in the [Code Contributor
+Workflow](https://cwiki.apache.org/confluence/display/DAFFODIL/Code+Contributor+Workflow),
+which you should read and follow carefully to ensure a successful code
+contribution to Apache Daffodil.
```

Comment:
These "You must" sentences might be easier to quickly parse (and more visible) if they were a list?

I think this also more belongs in a CONTRIBUTING.md file, but it's fine here for now. Might be a good future change though.